source/MRMesh/MRClosestWeightedPoint.h

```diff
@@ -51,11 +51,15 @@ struct MeshPointAndDistance
         return ( bidirectionalOrOutside ? eucledeanDist : -eucledeanDist ) - w;
     }
 
-    /// this distance is used internally to find the best surface point, which has the smallest inner distance;
-    /// innerDist() grows in both directions of the surface unlike weightedDist()
-    [[nodiscard]] float innerDist() const
+    /// comparison telling which point is closer to the location
+    auto operator <=> ( const MeshPointAndDistance& other ) const
     {
-        return eucledeanDist + ( bidirectionalOrOutside ? -w : w );
+        if ( bidirectionalOrOutside && other.bidirectionalOrOutside  )
+            return eucledeanDist - w <=> other.eucledeanDist - other.w;
+        if ( bidirectionalOrOutside != other.bidirectionalOrOutside )
+            return eucledeanDist <=> other.eucledeanDist;
```

Comment:
We completely ignore weights here?

Reply:
Yes, when we have two candidates such that according to one of them the point is outside, and according to the other it is inside, we should take the candidate that is actually closer to the point. Regardless of the weights of the points, only the closest point on the surface estimates inside/outside correctly (because we use pseudonormal to do this).